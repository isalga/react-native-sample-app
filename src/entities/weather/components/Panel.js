// @flow
import type { Weather } from '../../types'
import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import { colors } from '../../../styles'

type Props = {
  weather: ?Weather,
  title: ?string
}

const WeatherPanel = ({ weather, title }: Props) => {
  if (!weather)
    return (
      <View style={styles.container}>
        <Text>Uuups... information was not found</Text>
      </View>
    )
  else
    return (
      <View style={styles.container}>
        <Text>{ title }</Text>
        <Text>temperature: { weather.temperature }</Text>
        <Text>wind: { weather.windDirection } / { weather.windSpeed }</Text>
        <Text>humidity: { weather.humidity }</Text>
        <Text>precipitation: { weather.precipitation }</Text>
      </View>
    )
}

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    padding: 20,
    backgroundColor: colors.widgetBackground,
    borderRadius: 4,
    borderWidth: 1,
    borderColor: colors.border
  }
})

export default WeatherPanel
