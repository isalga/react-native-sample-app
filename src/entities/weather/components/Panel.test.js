import React from 'react'
import ReactDOM from 'react-dom'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Panel from './Panel'

Enzyme.configure({ adapter: new Adapter() })

const weather = {
  cityId: '1',
  temperature: 'testtemperature',
  windDirection: 'testwindDirection',
  windSpeed: 'testwindSpeed',
  humidity: 'testhumidity',
  precipitation: 'testprecipitation'
}

 const title = 'Berlin'

describe('Panel screen component', () => {

  it('should render correctly', () => {
    const wrapper = Enzyme.shallow(
      <Panel
        title={title}
        weather={weather}/>)
    expect(wrapper).toMatchSnapshot()
  })
})
