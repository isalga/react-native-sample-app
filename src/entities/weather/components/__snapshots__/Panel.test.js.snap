// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel screen component should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "gainsboro",
      "borderRadius": 4,
      "borderWidth": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Berlin
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    temperature: 
    testtemperature
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    wind: 
    testwindDirection
     / 
    testwindSpeed
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    humidity: 
    testhumidity
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    precipitation: 
    testprecipitation
  </Text>
</View>
`;
